tests/components/sentry/test_init.py

```diff
@@ -0,0 +1,330 @@
+    """Test if channel detection works from Home Assistant version number."""
+    assert get_channel(version) == channel
+
+
+async def test_process_before_send(hass: HomeAssistant):
+    """Test regular use of the Sentry process before sending function."""
+    hass.config.components.add("puppies")
+    hass.config.components.add("a_integration")
+
+    # These should not show up in the result.
+    hass.config.components.add("puppies.light")
+    hass.config.components.add("auth")
+
+    result = process_before_send(
+        hass,
+        options={},
+        channel="test",
+        huuid="12345",
+        system_info={"installation_type": "pytest"},
+        custom_components=["ironing_robot", "fridge_opener"],
+        event={},
+        hint={},
+    )
+
+    assert result
+    assert result["tags"]
+    assert result["contexts"]
+    assert result["contexts"]
+
+    ha_context = result["contexts"]["Home Assistant"]
+    assert ha_context["channel"] == "test"
+    assert ha_context["custom_components"] == "fridge_opener\nironing_robot"
+    assert ha_context["integrations"] == "a_integration\npuppies"
+
+    tags = result["tags"]
+    assert tags["channel"] == "test"
+    assert tags["uuid"] == "12345"
+    assert tags["installation_type"] == "pytest"
+
+
+async def test_event_with_platform_context(hass: HomeAssistant):
+    """Test extraction of platform context information during Sentry events."""
+
+    current_platform_mock = Mock()
+    current_platform_mock.get().platform_name = "hue"
+    current_platform_mock.get().domain = "light"
+
+    with patch(
+        "homeassistant.components.sentry.entity_platform.current_platform",
+        new=current_platform_mock,
+    ):
+        result = process_before_send(
+            hass,
+            options={},
+            channel="test",
+            huuid="12345",
+            system_info={"installation_type": "pytest"},
+            custom_components=["ironing_robot"],
+            event={},
+            hint={},
+        )
+
+    assert result
+    assert result["tags"]["integration"] == "hue"
+    assert result["tags"]["platform"] == "light"
+    assert result["tags"]["custom_component"] == "no"
+
+    current_platform_mock.get().platform_name = "ironing_robot"
+    current_platform_mock.get().domain = "switch"
+
+    with patch(
+        "homeassistant.components.sentry.entity_platform.current_platform",
+        new=current_platform_mock,
+    ):
+        result = process_before_send(
+            hass,
+            options={CONF_EVENT_CUSTOM_COMPONENTS: True},
+            channel="test",
+            huuid="12345",
+            system_info={"installation_type": "pytest"},
+            custom_components=["ironing_robot"],
+            event={},
+            hint={},
+        )
+
+    assert result
+    assert result["tags"]["integration"] == "ironing_robot"
+    assert result["tags"]["platform"] == "switch"
+    assert result["tags"]["custom_component"] == "yes"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6042836)